### PR TITLE
Pay Option for The Failed Order

### PR DESCRIPTION
**Pull request by shewa12**
The pay button will appear only if a user tries to make a payment and fails. Not applicable for the Manual Enrollment.

1. Now students will be able to make the payment for failed order
2. `tutor_redirect_after_payment` new function added
3. The purchase history table updated
4. **Back to checkout**  URL set using `wp_get_referer `

### CODE REVIEW

File: templates/dashboard/purchase_history.php [hunk 1]
```diff
@@ -163,15 +165,28 @@
 										<?php echo esc_html( tutor_get_formatted_price( $order->total_price ) ); ?>
 									</div>
 								</td>
-								<?php if ( Ecommerce::MONETIZE_BY === $monetize_by ) : ?>
+								<td>
+								<?php
+									echo wp_kses_post( tutor_utils()->translate_dynamic_text( $order->order_status, true ) );
+								?>
+								</td>
 								<td>
```
Comment (harunollyo) [inline templates/dashboard/purchase_history.php:173]: Monetized by tutor check missing for payment method column value

File: ecommerce/CheckoutController.php [hunk 2]
```diff
@@ -836,6 +826,38 @@ private function set_pay_now_alert_msg( $order_data ) {
...
+	 * Redirect to the payment gateway to complete the order
+	 * After completing the process it will redirect user to
+	 * order placement page
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function pay_incomplete_order() {
+		$order_id = Input::get( 'order_id', 0, Input::TYPE_INT );
```
Comment (harunollyo) [inline ecommerce/CheckoutController.php:841]:
```
if ( 'tutor_pay_incomplete_order' !== Input::post( 'tutor_action' ) {
  return;
}
```

Nonce check missing for tutor_action hook callback
`tutor_utils()->checking_nonce();`

File: templates/dashboard/purchase_history.php [hunk 3]
```diff
@@ -163,15 +165,28 @@
...
 									<div class="tutor-fs-7">
 										<?php echo esc_html( ucwords( $order->payment_method ?? '' ) ); ?>
 									</div>
 								</td>
-								<?php endif; ?>
 								<td>
-								<?php echo wp_kses_post( tutor_utils()->translate_dynamic_text( $order->order_status, true ) ); ?>
+								<form method="get">
+									<input type="hidden" name="tutor_action" value="tutor_pay_incomplete_order">
```
Comment (harunollyo) [inline templates/dashboard/purchase_history.php:180]: Add nonce field
```
<?php wp_nonce_field( tutor()->nonce_action, tutor()->nonce ); ?>
```

**Review by harunollyo — CHANGES_REQUESTED** [review]

File: templates/dashboard/purchase_history.php (hunk 3 above)
Comment (shewa12) [reply templates/dashboard/purchase_history.php:181]: `tutor_nonce_field` used

**Review by harunollyo — APPROVED** [review]

**Pull request merged**